### PR TITLE
PDA-49 get process form

### PR DESCRIPTION
**Pull request by BrenoQVDS**
Adding the Form deserializer and tests

### CODE REVIEW

**Comment by entando-service** [comment]
Something went wrong. Investigate!

**Comment by entando-service** [comment]
Something went wrong. Investigate!

**Comment by entando-service** [comment]
Something went wrong. Investigate!

**Comment by entando-service** [comment]
Something went wrong. Investigate!

**Comment by entando-service** [comment]
Something went wrong. Investigate!

**Comment by entando-service** [comment]
Something went wrong. Investigate!

**Comment by entando-service** [comment]
Something went wrong. Investigate!

File: src/test/java/org/entando/plugins/pda/pam/service/process/KieProcessFormServiceTest.java [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+import org.drools.core.io.impl.ClassPathResource;
+import org.entando.plugins.pda.core.model.form.Form;
+import org.entando.plugins.pda.core.model.form.FormField;
+import org.entando.plugins.pda.core.model.form.FormFieldInteger;
+import org.entando.plugins.pda.core.model.form.FormFieldText;
+import org.entando.plugins.pda.core.model.form.FormFieldType;
+import org.junit.Assert;
+import org.junit.Test;
+
+@SuppressWarnings("PMD.ExcessiveMethodLength")
```
Comment (ffleandro) [inline src/test/java/org/entando/plugins/pda/pam/service/process/KieProcessFormServiceTest.java]: Fix this PMD issue.
SuppressWarnings is not applicable to this issue.

File: src/test/java/org/entando/plugins/pda/pam/service/process/KieProcessFormServiceTest.java [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+public class KieProcessFormServiceTest {
+
+    @Test
+    public void deserializeProcessFormJson() {
```
Comment (ffleandro) [inline src/test/java/org/entando/plugins/pda/pam/service/process/KieProcessFormServiceTest.java]: This tests belongs to `KieProcessFormDeserializerTest`.
Check `KieTaskServiceTest` to see the kind of unit tests that should be implemented on this class.

File: src/test/java/org/entando/plugins/pda/pam/service/process/KieProcessServiceTest.java [hunk 3]
```diff
@@ -39,6 +39,7 @@
 import org.springframework.test.web.client.MockRestServiceServer;
 import org.springframework.web.client.RestTemplate;
 
+@SuppressWarnings("PMD.ExcessiveImports")
```
Comment (ffleandro) [inline src/test/java/org/entando/plugins/pda/pam/service/process/KieProcessServiceTest.java]: Not applicable.
If you didn't change this Class, why do you need to ignore this error?

**Comment by entando-service** [comment]
Something went wrong. Investigate!

**Comment by entando-service** [comment]
Something went wrong. Investigate!

**Comment by entando-service** [comment]
Something went wrong. Investigate!

**Comment by entando-service** [comment]
Something went wrong. Investigate!

File: src/test/java/org/entando/plugins/pda/pam/service/process/KieFormDeserializerTest.java [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+        SimpleModule module = new SimpleModule();
+        module.addDeserializer(Form.class, new KieFormDeserializer());
+        mapper.registerModule(module);
+
+        try {
+            List<Form> result = mapper.readValue(readFromFile("process-form.json"), new TypeReference<List<Form>>() {});
+            assertThat(result).isEqualTo(KieProcessFormTestHelper.createProcessForms());
+        } catch (IOException e) {
+            Assert.fail();
+        }
```
Comment (ffleandro) [inline src/test/java/org/entando/plugins/pda/pam/service/process/KieFormDeserializerTest.java]: try/catch not necessary, just add test signature to throw Exception

**Review by ffleandro — DISMISSED** [review]

**Comment by entando-service** [comment]
Something went wrong. Investigate!

**Review by ffleandro — DISMISSED** [review]

**Review by ffleandro — APPROVED** [review]

**Comment by entando-service** [comment]
All is well

**Pull request merged**